lib/ansible/modules/network/nxos/nxos_bfd_global.py

```diff
@@ -0,0 +1,308 @@
+                    'supported_by': 'network'}
+
+import datetime
+def logit(msg):
+    with open('/tmp/alog.txt', 'a') as of:
+        d = datetime.datetime.now().replace(microsecond=0).isoformat()
+        of.write("---- %s ----\n%s\n" % (d,msg))
+
+DOCUMENTATION = '''
+---
+module: nxos_bfd_global
+extends_documentation_fragment: nxos
+version_added: "2.9"
+short_description: Bidirectional Forwarding Detection (BFD) global-level configuration
+description:
+  - Manages Bidirectional Forwarding Detection (BFD) global-level configuration.
+
+author: Chris Van Heuveln (@chrisvanheuveln)
+notes:
+options:                 # **** To Be updated after initial review *****
+  echo_interface:
+    description:
+      - does a thing
+    required: false
+  echo_rx_interval:
+    description:
+      - does a thing
+    required: false
+  interval:
+    # <state> bfd interval <interval> min_rx <min_rx> multiplier <multiplier>
+    description:
+      - does a thing
+    required: false
+  slow_timer:
+    description:
+      - does a thing
+    required: false
+  startup_timer:
+    description:
+      - does a thing
+    required: false
+  afi specific commands (not available for 5k/6k)
+    ipv4_echo_rx_interval:
+    ipv6_echo_rx_interval:
+    description:
+      - does a thing
+    required: false
+    ipv4_interval:
+    ipv6_interval:
+    description:
+      - does a thing
+    required: false
+    ipv4_slow_timer:
+    ipv6_slow_timer:
+    description:
+      - does a thing
+    required: false
+  fabricpath-specific commands (only available for 5k/6k/7k)
+    fabricpath_interval:
+    description:
+      - does a thing
+    required: false
+    fabricpath_slow_timer:
+    description:
+      - does a thing
+    required: false
+    fabricpath_vlan:
+    description:
+      - does a thing
+    required: false
+'''
+EXAMPLES = '''
+- nxos_bfd_global:
+    echo_interface: Ethernet1/2
+    echo_interval: [50, 50, 3]
+    state: present
+'''
+
+RETURN = '''
+commands:
+    description: commands sent to the device
+    returned: always
+    type: list
+    sample: ["router bgp 65535", "vrf test", "router-id 192.0.2.1"]   # ***** TBD ****
+'''
+
+
+import re, yaml
+
+from ansible.module_utils.network.nxos.nxos import nxos_argument_spec, check_args
+from ansible.module_utils.network.nxos.nxos import load_config, run_commands, get_capabilities
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.common.config import CustomNetworkConfig
+from ansible.module_utils.connection import ConnectionError
+
+BFD_CMD_REF = '''
+# BFD does not support json format yet; use cli config strings instead.
+---
+echo_interface:
+  type: str
+  getval: (no )*bfd echo-interface *(\S+)*$
+  setval: '{0}bfd echo-interface {1}'
+  default: ''
+
+echo_rx_interval:
+  type: int
+  getval: bfd echo-rx-interval (\d+)$
+  setval: bfd echo-rx-interval {0}
+  default: 50
+  N3K: 250
+
+interval:
+  type: int_list
```

Comment:
I wonder if int_list is too specific and you should have a generic `list` type that can intermix ints and strs instead? Consider:

`route-map (\w+) permit (\d+)`
`address-family (\w+) autonomous-system (\d+)`
etc.

Reply:
My original intent was to also have `int_str` so that I could normalize the list contents accordingly. 
Instead I'll just normalize every list to str.